main.tf

```diff
@@ -97,12 +107,12 @@ module "jump" {
 
   count             = var.create_jump_vm ? 1 : 0
   name              = "${var.prefix}-jump"
-  azure_rg_name     = module.resource_group.name
+  azure_rg_name     = local.aks_rg.name
   azure_rg_location = var.location
   vnet_subnet_id    = module.vnet.subnets["misc"].id
   machine_type      = var.jump_vm_machine_type
-  azure_nsg_id      = module.nsg.id
-  tags              = module.resource_group.tags
+  azure_nsg_id      = local.nsg.id
+  tags              = var.tags
```

Comment:
So between this item and line 150 below we use `vars` for some items and `locals` for others. Is there a reason it's not more consistent? Either one or the other. Seems like the logic flow is a bit scattered here.

Reply:
generally, if the vars can be used as-is, without further logic, we use it.
If we need to add conditionals (there are - unfortunately - some dependencies between input vars), or wrapping logic (like data vs. resource for the byo items), we tend to use locals.